### PR TITLE
non-const versions only needed when using inlined ARM assembly in the module.

### PR DESCRIPTION
**Pull request by kaleb-himes**

### CODE REVIEW

**Review by SparkiDev — APPROVED** [review]

**Comment by SparkiDev** [comment]
Assuming you have tried newer GCC compilers and it worked ...

**Comment by kaleb-himes** [comment]
NOTE: const-poisoning was introduced after inline of ARM assembly for FIPS module which led to the seperation. Correct solution here is to guard non-const sections in the header with both `HAVE_FIPS_VERSION == 2` and `WOLFSSL_SP_ARM32_ASM` and `WOLFSSL_SP_ARM64_ASM`.

TODO: Refactor PR.

**Review by SparkiDev — APPROVED** [review]

File: wolfssl/wolfcrypt/sp.h [hunk 1]
```diff
@@ -105,13 +107,15 @@ WOLFSSL_LOCAL int sp_RsaPrivate_4096(const byte* in, word32 inLen,
     const mp_int* dqm, const mp_int* qim, const mp_int* mm, byte* out,
     word32* outLen);
 
-#endif /* HAVE_FIPS_VERSION && HAVE_FIPS_VERSION == 2 */
+#endif /* HAVE_FIPS_VERSION && HAVE_FIPS_VERSION == 2 && !gcc 4.x */
```
Comment (SparkiDev) [inline wolfssl/wolfcrypt/sp.h:110]: Fix comments to say ARM ASM.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: wolfssl/wolfcrypt/sp.h (hunk 1 above)
Comment (kaleb-himes) [reply wolfssl/wolfcrypt/sp.h]: Fixed, thank you @SparkiDev!

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**